### ISSUE TITLE
sam local doesn't work on OSX, get error Error: Running AWS SAM projects locally requires Docker. Have you got it installed?

### ISSUE THREAD

**Issue by websocketdev** [awslabs/aws-sam-cli#575]
sam local on OSX doesn't work with Docker even if Docker is installed. 

Latest version of docker is installed
Latest version of Node.JS is installed
When trying to run $ sam local start-lambda, I get the error:
Error: Running AWS SAM projects locally requires Docker. Have you got it installed?
Tried to set DOCKER_HOST to various things found around the internet, none actually work. 
There's an Ubuntu solution on Stack Overflow, but I cannot figure out how to apply it to the OSX version. https://stackoverflow.com/questions/50791354/running-aws-sam-projects-locally-get-error

It's not obvious at all how to get this to work on OSX. Surely this must be a common problem with an easy solution.

**Comment by jfuss** [comment]
@websocketdev Is docker running on your machine?

**Comment by reesezee** [comment]
I'm having the same problem on 
CentOS Linux release 7.5.1804 (Core) 
Python 2.7.5
Docker version 1.13.1, build dded712/1.13.1
active (running) since Tue 2018-08-21 15:34:20 CDT; 19h ago

Am I supposed to be running sam local as root ?
or does my user need to be in the docker group to launch containers, 
because with out root permissions I dont see how you're going to launch the container

**Comment by sanathkr** [comment]
Yes, your user must have access to Docker daemon in order for SAM CLI to create/manage containers.

**Comment by jvidin** [comment]
I have an old Mac mini where newer versions of 'Docker for Mac' does not work , so I'm stuck on 'Docker Machine' Im receiving the message `Error: Running AWS SAM projects locally requires Docker. Have you got it installed?
` I can easy play with docker from terminal and its working . Can anyone confirm 'Docker Machine' is suitable for running SAM locally ?

**Comment by m-denton** [comment]
FWIW I ran into the same issue. Ended up clearing everything out, restarting the daemon/computer, etc. Ended up being a bad volume binding for 
`-v /var/run/docker.sock:/var/run/docker.sock`

**Comment by anildharra** [comment]
I have cross checked that Docker is already installed on my machine, but still getting below error while running "sam local start-api"

Error: Running AWS SAM projects locally requires Docker. Have you got it installed?

**Comment by reesezee** [comment]
You need to be root.
Try sudo

**Comment by anildharra** [comment]
Hi 
sudo is not supported in win7. What may be the alternative for this type of issue.

Thanks
Anil

**Comment by awood45** [comment]
I'd be curious how you've installed Docker - it currently lists Windows 10 as a requirement, so you possibly need to upgrade Windows and reinstall. Otherwise, would recommend reaching out to Docker support channels like their forums to look at what the issue might be with your installation.

I do not think sudo/root is required at all to use Docker if you install it correctly, I don't use sudo/root for Docker on any machine I use.

**Comment by reesezee** [comment]
Docker doesn’t run on a MAC without sudo unless you do specific nonstandard
steps after install.

**Comment by reesezee** [comment]
Same with Linux.

**Comment by anildharra** [comment]
Hi @Alex Wood awood45 
Let see the grab below which clearly mentioned that I have Docker installed on Win7.

![image](https://user-images.githubusercontent.com/17697925/73576338-3bfb4300-44a0-11ea-9ecd-493df924277e.png)

Re-iterating the issue.
D:\anil\work\learning\amazon\projects\sam\hello-world-sam-app\aws-sam-app>sam local start-api

Error: Running AWS SAM projects locally requires Docker. Have you got it installed?

**Comment by anildharra** [comment]
Lets see another grab. [Docker]

: where docker

![image](https://user-images.githubusercontent.com/17697925/73576604-d491c300-44a0-11ea-8adc-3ac1ece27cd1.png)

**Comment by awood45** [comment]
Interesting - what do you get when you run `docker ps`?

Are you able to identify your version of Docker?

**Comment by awood45** [comment]
We do have an [installation guide](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install-windows.html) for Windows, and it's possible the troubleshooting steps there may work for you. If Docker is on your path and you're getting this error, that would be a new issue.

**Comment by awood45** [comment]
It also is possible you have Docker installed, but it's not running. Docker Desktop is more likely to start up on its own, it's possible you need to do extra steps to ensure Docker Toolbox is actually running before running SAM commands.

**Comment by anildharra** [comment]
The docker version is here in grab:

![image](https://user-images.githubusercontent.com/17697925/73577193-461e4100-44a2-11ea-895a-8598fac3dd81.png)

I have deployed a sample app at AWS successfully here is the link:
https://2ns4r7wi31.execute-api.us-east-1.amazonaws.com/Prod/hello/

Now I want to test it locally, so getting above mentioned Error. That is my problem.
Suggest any clue.

**Comment by awood45** [comment]
I suspect at this point Docker isn't running - you want to run `docker ps` to see it running, or otherwise make sure Docker is open when running SAM commands. Docker Desktop and Docker for OSX tend to be proactive about being running, but if you have Docker installed, yet not running, that is the error you see.

**Comment by anildharra** [comment]
Hi Awood45,

After running two things below, then app start working locally.
![image](https://user-images.githubusercontent.com/17697925/73577937-63540f00-44a4-11ea-8567-9f68bcb68e36.png)

But while running/testing app locally it shows below mentioned MESSAGE.

![image](https://user-images.githubusercontent.com/17697925/73577992-90082680-44a4-11ea-8651-10381c6eb49f.png)

Am missing somethnig.

**Comment by awood45** [comment]
Try adding `/hello` to the end of that path, if you're doing the Hello World example.

**Comment by anildharra** [comment]
Hi Awood45,
As per your instruction, I added "hello" to the end. Now it shows me another error.
Any clue.
![image](https://user-images.githubusercontent.com/17697925/73578428-a1056780-44a5-11ea-8964-e0411f93910c.png)

**Comment by awood45** [comment]
Would depend on how you ran init and if you've made changes. If you'd like, open a separate GitHub issue to discuss. I would recommend, now that you have running Docker figured out, checking the [getting started guide](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-getting-started.html) in general.

**Comment by anildharra** [comment]
Thanks a lot, Awood45 for your sincere guidance and instructions. You resolved things better and I am able to the next level of running the app locally. I will create another Github request for the above-mentioned Issue.
Thanks again.
